tests/potential/test_pair.py

```diff
@@ -97,6 +97,173 @@ def test_derivative(self):
         with self.assertRaises(ValueError):
             u = lj._derivative(param='simga', r=r_input, epsilon=1.0, sigma=1.0)
 
+class test_Spline(unittest.TestCase):
+    """Unit tests for pair.Spline"""
+
+    def test_init(self):
+        """Test creation from data"""
+        #test diff mode
+        s = pair.Spline(types=('1',), num_knots=3)
+        self.assertEqual(s.num_knots, 3)
+        self.assertEqual(s.mode, 'diff')
+        coeff = potential.CoefficientMatrix(types=('1',),
+                                            params=('r-0','r-1','r-2','knot-0','knot-1','knot-2','rmin','rmax','shift'),
+                                            default={'rmin':False,'rmax':False,'shift':False})
+        self.assertCountEqual(s.coeff.types, coeff.types)
+        self.assertCountEqual(s.coeff.params, coeff.params)
+        self.assertDictEqual(s.coeff.default.todict(), coeff.default.todict())
+
+        #test value mode
+        s = pair.Spline(types=('1',), num_knots=3, mode='value')
+        self.assertEqual(s.num_knots, 3)
+        self.assertEqual(s.mode, 'value')
+        coeff = potential.CoefficientMatrix(types=('1',),
+                                            params=('r-0','r-1','r-2','knot-0','knot-1','knot-2','rmin','rmax','shift'),
+                                            default={'rmin':False,'rmax':False,'shift':False})
+        self.assertCountEqual(s.coeff.types, coeff.types)
+        self.assertCountEqual(s.coeff.params, coeff.params)
+        self.assertDictEqual(s.coeff.default.todict(), coeff.default.todict())
+
+
+        #test invalid number of knots
+        with self.assertRaises(ValueError):
+            s = pair.Spline(types=('1',), num_knots=1)
+
+        #test invalid mode
+        with self.assertRaises(ValueError):
+            s = pair.Spline(types=('1',), num_knots=3, mode='val')
+
+    def test_from_array(self):
+        """Test from_array method"""
+        r_arr = [1,2,3]
+        u_arr = [9,4,1]
+
+        #test diff mode
+        s = pair.Spline(types=('1',), num_knots=3)
+        s.from_array(pair=('1','1'), r=r_arr, u=u_arr)
+
+        self.assertEqual(s.coeff['1','1']['r-0'].value, 1)
```

Comment:
I would maybe access these with the generator or with `_knot_params` so that you don't have to hardcode the parameter names.